src/collaboration/push-protocol.js

```diff
@@ -220,8 +221,11 @@ module.exports = class PushProtocol {
       // If the remote sent us its clock, update our local copy
       if (newRemoteClock) {
         let clock
-        if (isPinner) {
-          // If the remote is a pinner, assume its clock is authoritative
+        // If the remote is a pinner, assume its clock is authoritative.
+        // If remote is asking us to start pushing, we're going to start
+        // from the remote clock.
+        const overwriteRemoteClock = isPinner || (!wasPushing && pushing)
```

Comment:
This makes sense, but should it be inside `queue.add()`? Otherwise it may get overwritten if the pusher is in the middle of pushing state

Reply:
@dirkmc I've addressed these, putting these updates in the queue (and clearing the queue beforehand). Could you re-review?

Reply:
LGTM 👍